src/brsTypes/components/RoSGNode.ts

```diff
@@ -231,6 +325,181 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
+    private addfields = new Callable("addfields", {
+        signature: {
+            args: [new StdlibArgument("fields", ValueKind.Object)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, fields: RoAssociativeArray) => {
+            if (!(fields instanceof RoAssociativeArray)) {
+                return BrsBoolean.False;
+            }
+
+            fields.getValue().forEach((value, key) => {
+                let fieldName = new BrsString(key);
+                let fieldType = ValueKind.toString(value.kind);
+                if (this.get(fieldName) === BrsInvalid.Instance) {
+                    this.set(fieldName, value);
+                }
+            });
+
+            return BrsBoolean.True;
+        },
+    });
+
+    /** Returns the value of the field passed as argument, if the field doesn't exist it returns invalid. */
+    private getfield = new Callable("getfield", {
+        signature: {
+            args: [new StdlibArgument("fieldname", ValueKind.String)],
+            returns: ValueKind.Dynamic,
+        },
+        impl: (interpreter: Interpreter, fieldname: BrsString) => {
+            return this.get(fieldname);
+        },
+    });
+
+    private observefield = new Callable("observefield", {
+        signature: {
+            args: [
+                new StdlibArgument("fieldname", ValueKind.String),
+                new StdlibArgument("functionname", ValueKind.String),
+            ],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, fieldname: BrsString) => {
+            return BrsBoolean.True;
+        },
+    });
+
+    /** Removes the given field from the node */
+    /** TODO: node built-in fields shouldn't be removable (i.e. id, change, focusable,) */
+    private removefield = new Callable("removefield", {
+        signature: {
+            args: [new StdlibArgument("fieldname", ValueKind.String)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, fieldname: BrsString) => {
+            if (this.get(fieldname) !== BrsInvalid.Instance) {
+                this.fields.delete(fieldname.value);
+            }
+
+            return BrsBoolean.True;
+        },
+    });
+
+    /** Updates the value of an existing field only if the types match. */
+    private setfield = new Callable("setfield", {
+        signature: {
+            args: [
+                new StdlibArgument("fieldname", ValueKind.String),
+                new StdlibArgument("value", ValueKind.Dynamic),
+            ],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, fieldname: BrsString, value: BrsType) => {
+            let field = this.get(fieldname);
+
+            if (field === BrsInvalid.Instance) {
+                return BrsBoolean.False;
+            }
+
+            if (
+                ValueKind.toString(field.kind).toLowerCase() !==
```

Comment:
When would these kinds get different casings?  Do they not all come from the same type-safe source?

Reply:
good call 🤦‍♂ , updating this